src/topology/locally_constant/basic.lean

```diff
@@ -426,4 +426,42 @@ lemma coe_desc {X α β : Type*} [topological_space X] (f : X → α) (g : α 
 
 end desc
 
+section indicator
+variables {R : Type*} [has_zero R] {U : set X} (f : locally_constant X R)
+
+/-- Given a clopen set `U` and a locally constant function `f`, `locally_constant.indicator`
+  returns the locally constant function that is `f` on `U` and `0` otherwise. -/
+@[simps]
+noncomputable def indicator (hU : is_clopen U) :
+  locally_constant X R :=
+{ to_fun := set.indicator U f,
+  is_locally_constant :=
+    begin
+      rw is_locally_constant.iff_exists_open, rintros x,
+      obtain ⟨V, hV, hx, h'⟩ := (is_locally_constant.iff_exists_open _).1 f.is_locally_constant x,
+      by_cases x ∈ U,
+      { refine ⟨U ∩ V, is_open.inter hU.1 hV, set.mem_inter h hx, _⟩, rintros y hy,
+        rw set.mem_inter_iff at hy, rw [set.indicator_of_mem hy.1, set.indicator_of_mem h],
+        apply h' y hy.2, },
+      { rw ←set.mem_compl_iff at h, refine ⟨Uᶜ, (is_clopen.compl hU).1, h, _⟩,
+        rintros y hy, rw set.mem_compl_iff at h, rw set.mem_compl_iff at hy,
+        simp [h, hy], },
+    end, }
+
+variables (a : X)
+open_locale classical
+
+theorem apply_indicator (hU : is_clopen U) :
```

Comment:
```suggestion
theorem indicator_apply (hU : is_clopen U) :
```

Reply:
Don't you think the lemma generated by `simps` is useful way more often than this lemma? There is a lot more (specific) API for `set.indicator` than `ite`.

Reply:
I agree; I don't think we need any of the API beyond the simps lemma, as it all follows by unfolding the bundling with dsimp, then applying the appropriate set lemma